controllers/storagecluster/cephnfs.go

```diff
@@ -0,0 +1,340 @@
+
+	switch {
+	case err == nil:
+		if existingBlockPool.DeletionTimestamp != nil {
+			r.Log.Info("Unable to restore CephBlockPool because it is marked for deletion.", "CephBlockPool", klog.KRef(existingBlockPool.Namespace, existingBlockPool.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to restore initialization object %s because it is marked for deletion", existingBlockPool.Name)
+		}
+
+		r.Log.Info("Restoring original CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+		existingBlockPool.ObjectMeta.OwnerReferences = cephBlockPool.ObjectMeta.OwnerReferences
+		existingBlockPool.Spec = cephBlockPool.Spec
+		err = r.Client.Update(context.TODO(), &existingBlockPool)
+		if err != nil {
+			r.Log.Error(err, "Failed to update CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+			return reconcile.Result{}, err
+		}
+	case errors.IsNotFound(err):
+		r.Log.Info("Creating CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+		err = r.Client.Create(context.TODO(), cephBlockPool)
+		if err != nil {
+			r.Log.Error(err, "Failed to create CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// ensureDeleted deletes the CephNFS related CephBlockPool resource owned by the StorageCluster
+func (obj *ocsCephNFSBlockPool) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	foundCephBlockPool := &cephv1.CephBlockPool{}
+	cephBlockPool, err := r.newCephNFSBlockPoolInstance(sc)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: cephBlockPool.Name, Namespace: sc.Namespace}, foundCephBlockPool)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Uninstall: CephBlockPool not found.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("uninstall: unable to retrieve CephBlockPool %v: %v", cephBlockPool.Name, err)
+	}
+
+	if cephBlockPool.GetDeletionTimestamp().IsZero() {
+		r.Log.Info("Uninstall: Deleting CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+		err = r.Client.Delete(context.TODO(), foundCephBlockPool)
+		if err != nil {
+			r.Log.Error(err, "Uninstall: Failed to delete CephBlockPool.", "CephBlockPool", klog.KRef(foundCephBlockPool.Namespace, foundCephBlockPool.Name))
+			return reconcile.Result{}, fmt.Errorf("uninstall: Failed to delete CephBlockPool %v: %v", foundCephBlockPool.Name, err)
+		}
+	}
+
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: cephBlockPool.Name, Namespace: sc.Namespace}, foundCephBlockPool)
```

Comment:
why do we need more one Get here?

Reply:
> why do we need more one Get here?

same explanation as commented here https://github.com/red-hat-storage/ocs-operator/pull/1635/files#r852009853

This is the pattern that is currently followed in the existing code.